docs/madr/decisions/008-mesh-logging.md

```diff
@@ -0,0 +1,381 @@
+# MeshLogging
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4733
+
+# Context and Problem Statement
+
+[New policy matching MADR](./005-policy-matching.md) introduces a new approach how Kuma applies configuration to proxies.
+Rolling out strategy implies creating a new set of policies that satisfy a new policy matching.
+Current MADR aims to define a `MeshLogging`.
+
+# Considered Options
+
+* Create a MeshLogging policy
+  * Keep backends in the `Mesh` resource for now, move them into `MeshLoggingBackend` policy in the future
+  * Move backend to the `MeshLoggingBackend` policy now
+  * Embed backend in `MeshLogging` policy
+
+# Decision Outcome
+
+To be determined.
+
+## Overview
+
+MeshLogging allows users to log incoming and outgoing traffic for services.
+This feature is useful in the following scenarios:
+* auditing - services that contain important data might want to log access to its resources for auditing purposes
+* debugging - an engineer might want to see traffic going in/out of a service to debug misbehaviour
+* insights - logs are useful when trying to understand users and their behaviour
+
+## Specification
+
+### Naming
+
+During the weekly meeting we started talking about the appropriate name for this policy, here are the names mentioned:
+- `MeshAccessLog`
+- `MeshLog`
+- `MeshTrafficLog`
+
+Please vote on your preferred name in the comments. Feel free to edit this document and add your suggestions.
+
+### Matching
+
+There are 3 parts to matching, each meaning a different part of the request flow:
+- `spec.targetRef` - where the policy attaches to, `to` and `from` are relative to this element(s)
+- `spec.to.targetRef` - outbound(s) of the `spec.targetRef`
+- `spec.from.targetRef` - inbound(s) of the `spec.targetRef`
+
+#### Top level
+
+`spec.targetRef` can have the following kinds: `Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute`
+
+Matching on `MeshGatewayRoute` and `MeshHTTPRoute` can be achieved by using a
+[HeaderMatcher](https://github.com/envoyproxy/envoy/blob/23a9a686bb4237934cd575d8e62d3e0df98b59ee/api/envoy/config/accesslog/v3/accesslog.proto#L207)
+with a value of `:path`.
+
+#### From level
+
+`spec.from.targetRef` can only have: `Mesh|MeshSubset|MeshService|MeshServiceSubset`.
+Matching on `MeshGatewayRoute` and `MeshHTTPRoute` does not make sense (there is no `route` that a request originates **from**).
```

Comment:
I think I mistakenly assumed that all of the Mesh / Service info ends up in `dynamic_metadata` so that it can be filtered by [metadata_filter](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/accesslog/v3/accesslog.proto#envoy-v3-api-msg-config-accesslog-v3-metadatafilter). I'm guessing it's not, so we would need to create a filter that would get that info from the cert and pass it to `dynamic_metadata`. This is extra work but this would solve the general case of using `MeshSubset|MeshService|MeshServiceSubset` as a target for HTTP/TCP. Without a filter of that kind we're never able to target these things in TCP right?

Reply:
We discussed live that we should be able to support any subset that can be extracted from the spiffe info by leveraging the accessLog filters.
In the case where mTLS is not enabled let's just ignore From matchers that are not `Mesh`. We'll need to figure out a way to expose this to the user in a nice way (we discussed a warning in the policy matching inspect API) but this is a separate issue.

Reply:
After chatting with @lobkovilya we decided that for the first iteration we will only support Mesh and for the general use case of targeting `MeshSubset|MeshService|MeshServiceSubset` we will implement this the following way (in a separate PR / issue):
- get the spiffe info from certificate in a custom Lua envoy filter (or use something predefined if exists) (you can get the info by calling `handle:streamInfo():downstreamSslConnection():uriSanPeerCertificate()` - see [example here](https://github.com/allegro/envoy-control/blob/087a8d3bf9b923f2013dabcf3adef2bfe2d9533f/envoy-control-core/src/main/resources/lua/ingress_client_name_header.lua#L13))
- pass spiffe info into dynamic metadata by calling `dynamicMetadata:set(filterName, key, value)` ([see docs](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter#set))
- use [metadata_filter](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/accesslog/v3/accesslog.proto#envoy-v3-api-field-config-accesslog-v3-accesslogfilter-metadata-filter) with data from the previous point